src/stdatamodels/jwst/transforms/models.py

```diff
@@ -2059,6 +2060,77 @@ def evaluate(self, x, y, x0, y0, order):
         return model(x, y, x0, y0, order)
 
 
+class MIRIWFSSBackwardDispersion(Model):
+    """
+    Calculate the dispersion extent of MIRI WFSS pixels.
+    """
+
+    standard_broadcasting = False
+    _separable = False
+    fittable = False
+    linear = False
+
+    n_inputs = 2
+    n_outputs = 4
+
+    def __init__(self, lmodels=None, xmodels=None, ymodels=None, name=None):
+        """
+        Initialize the model.
+
+        Parameters
+        ----------
+        lmodels : list
+            The list of models for the polynomial model in l
+        xmodels : list[tuple]
+            The list of tuple(models) for the polynomial model in x
+        ymodels : list[tuple]
+            The list of tuple(models) for the polynomial model in y
+        name : str, optional
+            Name of the model
+        """
+        self.xmodels = xmodels
+        self.ymodels = ymodels
+        self.lmodels = lmodels
+
+        if name is None:
+            name = "miriwfss_backward_dispersion"
+        super(MIRIWFSSBackwardDispersion, self).__init__(name=name)
+        self.inputs = ("x", "y")
+        self.outputs = ("x", "y", "x0", "y0")
```

Comment:
How do these models avoid taking wavelength as an input?  Surely the wavelength is what allows you to go between the direct image x,y and the grism x,y?

Reply:
There is only 1 order. I can add order to the model but it is always = 1. 
The lmodels (lambda models) is how to determine the wavelength. We do not provide a wavelength when going to direct to dispersed because we have not information on the wavelength in the direct image. But we can calculate the wavelength in the dispersed using the lmodels.

Reply:
But you would still need a wavelength to get a _specific_ dispersed value.  If I have a direct image, and I want to model a source with a given spectrum, I need to know which part of the trace corresponds to what wavelength.

Right now the `evaluate` method you wrote has `t` hard-coded to `np.linspace(0, 1, 10)`, and just gives me the x,y values at all 10 of those trace positions.  In the other models, NIRISS and NIRCam, `t` is determined from the input wavelength and the lmodel, and then used to get x,y.  For those models, if I pass in one wavelength, I get out one x,y position. If I pass in 100 wavelengths, I get out 100 x,y positions.